RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelDigisClustersFromSoA.cc

```diff
@@ -117,7 +121,7 @@ void SiPixelDigisClustersFromSoA::produce(edm::StreamID, edm::Event& iEvent, con
     for (int32_t ic = 0; ic < nclus + 1; ++ic) {
       auto const& acluster = aclusters[ic];
       // in any case we cannot  go out of sync with gpu...
-      if (acluster.charge < clusterThreshold and !isPhase2_)
+      if (acluster.charge < clusterThreshold and !std::is_same_v<TrackerTraits, pixelTopology::Phase2>)
```

Comment:
```suggestion
      if (not std::is_same_v<TrackerTraits, pixelTopology::Phase2> and acluster.charge < clusterThreshold)
```
The conditions are evaluated in order, so I'd suggest moving the compile-time one to be first.